src/utils/parser/scopes.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import toPairs from "lodash/toPairs";
+import uniq from "lodash/uniq";
+
+import type { Scope } from "./types";
+
+function getScopeVariables(scope: Scope) {
+  const { bindings } = scope;
+
+  return toPairs(bindings).map(([name, binding]) => ({
+    name,
+    references: binding.referencePaths
+  }));
+}
+
+function getScopeChain(scope: Scope): Scope[] {
+  const scopes = [scope];
+
+  do {
+    scopes.push(scope);
+  } while ((scope = scope.parent));
```

Comment:
I think there might be an issue here.. though I might be missing something, because I see that its used in other bits of code. The issue is, I am not sure how this loop terminates -- from what I understand reading this, the expression in the while loop returns `undefined`, which is the same whether `scope.parent` is defined or not, and that can be done endlessly. When I tried running just this code with some fake data I did get stuck in an infinite loop.

So two things -- perhaps we can make this condition safer? and, why the heck is it working??